nominatim/tools/tiger_data.py

```diff
@@ -0,0 +1,114 @@
+"""
+Functions for importing tiger data and handling tarbar and directory files
+"""
+import logging
+import os
+import tarfile
+import selectors
+
+from ..db.connection import connect
+from ..db.async_connection import DBConnection
+from ..db.sql_preprocessor import SQLPreprocessor
+
+
+LOG = logging.getLogger()
+
+
+def handle_tarfile_or_directory(data_dir):
+    """ Handles tarfile or directory for importing tiger data
+    """
+
+    tar = None
+    if data_dir.endswith('.tar.gz'):
+        tar = tarfile.open(data_dir)
+        sql_files = [i for i in tar.getmembers() if i.name.endswith('.sql')]
+        LOG.warning("Found %d SQL files in tarfile with path %s", len(sql_files), data_dir)
+        if not sql_files:
+            LOG.warning("Tiger data import selected but no files in tarfile's path %s", data_dir)
+            return None, None
+    else:
+        files = os.listdir(data_dir)
+        sql_files = [os.path.join(data_dir, i) for i in files if i.endswith('.sql')]
+        LOG.warning("Found %d SQL files in path %s", len(sql_files), data_dir)
+        if not sql_files:
+            LOG.warning("Tiger data import selected but no files found in path %s", data_dir)
+            return None, None
+
+    return sql_files, tar
+
+
+def handle_threaded_sql_statements(sel, file):
+    """ Handles sql statement with multiplexing
+    """
+
+    lines = 0
+    end_of_file = False
+    # Using pool of database connections to execute sql statements
+    while True:
+        if end_of_file:
+            break
+        for key, _ in sel.select(1):
+            conn = key.data
+            try:
+                if conn.is_done():
+                    sql_query = file.readline()
+                    lines += 1
+                    if not sql_query:
+                        end_of_file = True
+                        break
+                    conn.perform(sql_query)
+                    if lines == 1000:
+                        print('. ', end='', flush=True)
+                        lines = 0
+            except Exception as exc: # pylint: disable=broad-except
+                LOG.error('Wrong SQL statement: %s', exc)
```

Comment:
Can you move this to level 'info'? We have silently ignored those errors already before.

Reply:
Sure